src/olympia/amo/middleware.py

```diff
@@ -458,3 +462,22 @@ def process_response(self, request, response):
     def process_exception(self, request, exception):
         if not isinstance(exception, Http404):
             statsd.incr('response.500')
+
+
+class GraphiteRequestTimingMiddleware(GraphiteRequestTimingMiddlewareUpstream):
+    """Like django-statsd's GraphiteRequestTimingMiddleware but handles class
+    based views correctly in Django 4.2, leveraging view_class attribute.
+
+    Inspired from Django's <HttpRequest>.resolver_match._func_path"""
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        view = view_func
+        if hasattr(view, 'view_class'):
+            view = view.view_class
+        if not hasattr(view, '__name__'):
+            # A class-based view
+            view = view.__class__
+
+        request._view_module = view.__module__
+        request._view_name = view.__name__
+        request._start_time = time.time()
```

Comment:
we don't have to `super()` I guess?

Reply:
We could call `super()` before our stuff but we'd end up overriding `request._view_module` and `request._view_name` so I didn't think it was worth it but maybe?

Reply:
🤷‍♂️ I just noticed it wasn't there. I guess there could be more in the upstream function in the future, but on the other hand it might be incompatible so... it's fine.